client/mm/exchange_adaptor.go

```diff
@@ -2195,64 +2195,72 @@ func (u *unifiedExchangeAdaptor) OrderFeesInUnits(sell, base bool, rate uint64)
 	return baseFeesInUnits + quoteFeesInUnits, nil
 }
 
-func (u *unifiedExchangeAdaptor) cancelAllOrders(ctx context.Context) {
-	doCancels := func(epoch *uint64, dexOrderBooked func(oid order.OrderID, sell bool) bool) bool {
-		u.balancesMtx.RLock()
-		defer u.balancesMtx.RUnlock()
+// tryCancelOrders cancels all booked DEX orders that are past the free cancel
+// threshold. If cancelCEXOrders is true, it will also cancel CEX orders. True
+// is returned if all orders have been cancelled. If cancelCEXOrders is false,
+// false will always be returned.
+func (u *unifiedExchangeAdaptor) tryCancelOrders(ctx context.Context, epoch *uint64, dexOrderBooked func(oid order.OrderID, sell bool) bool, cancelCEXOrders bool) bool {
+	u.balancesMtx.RLock()
+	defer u.balancesMtx.RUnlock()
 
-		done := true
+	done := true
 
-		freeCancel := func(orderEpoch uint64) bool {
-			if epoch == nil {
-				return true
-			}
-			return *epoch-orderEpoch >= 2
+	freeCancel := func(orderEpoch uint64) bool {
+		if epoch == nil {
+			return true
 		}
+		return *epoch-orderEpoch >= 2
+	}
 
-		for _, pendingOrder := range u.pendingDEXOrders {
-			o := pendingOrder.currentState().order
+	for _, pendingOrder := range u.pendingDEXOrders {
+		o := pendingOrder.currentState().order
 
-			var oid order.OrderID
-			copy(oid[:], o.ID)
-			// We need to look in the order book to see if the cancel succeeded
-			// because the epoch summary note comes before the order statuses
-			// are updated.
-			if !dexOrderBooked(oid, o.Sell) {
-				continue
-			}
+		var oid order.OrderID
+		copy(oid[:], o.ID)
+		// We need to look in the order book to see if the cancel succeeded
+		// because the epoch summary note comes before the order statuses
+		// are updated.
```

Comment:
We use a `core.ResolvedEpoch` note now, so this is no longer needed.

Reply:
Fixed.